scripts/in_progress/regrid.py

```diff
@@ -0,0 +1,405 @@
+    Input: WRF / ERA5 / CERES / Rutgers xarray dataset/dataarray,
+           WRF projection, coordinate system 
+    Output: xarray dataset/dataarray in WRF24 grid
+    """
+    # Transform to WRF Projection
+    xform_pts = wrf_proj['cart_proj'].transform_points(wrf_proj['wrf_crs'],to_np(ds.lon.values),to_np(ds.lat.values))
+    wrf_x = xform_pts[...,0]
+    wrf_y = xform_pts[...,1]
+    ds_wrf=ds.assign_coords({'lat':(('south_north','west_east'),wrf_y),
+                                                       'lon':(('south_north','west_east',),wrf_x)})
+    # update attrs
+    ds_wrf.attrs['output_grid'] = 'wrf24'
+    return ds_wrf
+
+def find_tem_freq(ds_in):
+    """
+    Functions: identify temporal frequency
+    Input: 
+    1. ds_in: xarray dataset/dataarray
+    Output:
+    !. temporal frequency as string
+    """
+    freq = xr.infer_freq(ds_in.time)
+    # if xr.infer_freq does not work
+    if freq is None:
+        delta = pd.to_timedelta(np.diff(ds_in.time)).mean()
+        if delta.days <=31 and delta.days>=28:
+            freq = 'MS'
+        elif delta.days == 7:
+            freq = 'W'
+    else:
+        pass
+    # return freq name based on results
+    if 'MS' in freq or 'ME' in freq:
+        ds_freq = 'monthly'
+    elif 'W' in freq:
+        ds_freq = 'weekly'
+    elif 'Q' in freq:
+        ds_freq = 'seasonal'
+    elif 'Y' in freq:
+        ds_freq = 'yearly'
+    return ds_freq
+
+# Function to build filename of exported netcdf
+def build_regridded_nc_filename(ds_in):
+    """
+    Function: build filename for exported regridded nc
+    Inputs:
+    1. ds_in: xarray dataset/dataarray of output dataset with ds_name, input_grid, output_grid, ds_path as attrs
+    Output:
+    Filename
+    """
+    # for WRF
+    if re.search('wrf', ds_in.attrs['ds_path'], re.I):
+        # for WRF norm
+        if re.search('-norm-', ds_in.attrs['ds_path']):
+            filename = ds_in.attrs['ds_path'].replace(ds_in.attrs['input_grid'],ds_in.attrs['output_grid'])
+        # for wrf monthly series
+        else:
+            # build path 
+            path, scen = build_parameter(ds_in)
+            # replace grid in path
+            path_new = path.replace(ds_in.attrs['input_grid'],ds_in.attrs['output_grid'])
+            # find period
+            start_year = str(ds_in.time[0].dt.year.values)
+            end_year = str(ds_in.time[-1].dt.year.values)
+            # get wrf subcategory
+            wrf_cat = get_nth_word_custom_delimiter(ds_in.attrs['description'], ' ', 1)
+            # name for netcdf
+            filename_sub = wrf_cat + '_' + scen + '_' + 'monthly-' + start_year + '-' + end_year +'.nc' 
+            # build filename
+            filename = os.path.join(path_new, filename_sub)
+    # for obs dataset 
+    elif re.search('Data', ds_in.attrs['ds_path']):
+        # for dataset built from multiple files
+        if re.search('\*.nc', ds_in.attrs['ds_path']):
```

Comment:
A simple way to check this would be `some_str[-3:].lower() == ".nc"`; regular expressions are only very rarely necessary. You can also do things like `if "WRF" in some_str.upper():` to check if the string "WRF" or any variant of its capitalization appears in the string.